cv-data-service-library/src/main/java/com/trihydro/library/helpers/DbInteractions.java

```diff
@@ -50,65 +71,85 @@ public Connection getConnectionPool() throws SQLException {
             }
             throw ex;
         }
     }
 
+    /**
+     * Executes the given PreparedStatement to perform an update or delete operation on the database.
+     *
+     * @param preparedStatement the PreparedStatement to execute for the update or delete operation
+     * @return true if the operation affected at least one row in the database, false otherwise
+     */
     public boolean updateOrDelete(PreparedStatement preparedStatement) {
-
         boolean result = false;
-
         try {
             if (preparedStatement.executeUpdate() > 0) {
                 result = true;
             }
         } catch (SQLException e) {
-            e.printStackTrace();
+            log.error("Error in updateOrDelete", e);
         }
-
         return result;
     }
 
+    /**
+     * Executes a delete operation using the provided PreparedStatement. The method
+     * returns a boolean indicating whether the operation was successful. If an
+     * exception occurs during execution, it is logged, and the method returns false.
+     *
+     * @param preparedStatement the PreparedStatement used to perform the delete operation
+     * @return true if the delete operation was successful, false if it failed
+     */
     public boolean deleteWithPossibleZero(PreparedStatement preparedStatement) {
         boolean result = false;
         try {
             preparedStatement.executeUpdate();
             result = true;
         } catch (SQLException e) {
-            e.printStackTrace();
+            log.error("Error in deleteWithPossibleZero", e);
         }
-
         return result;
     }
 
+    /**
+     * Executes the given prepared statement, logs the generated key if available, and returns the key.
+     *
+     * @param preparedStatement the prepared statement to be executed
+     * @param type              the type of operation or entity being logged, for logging purposes
+     * @return the generated key as a Long if the execution is successful and a key is generated, otherwise null
+     */
     public Long executeAndLog(PreparedStatement preparedStatement, String type) {
         Long id = null;
         try {
             if (preparedStatement.executeUpdate() > 0) {
-                ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
-                try {
+                try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
                     if (generatedKeys != null && generatedKeys.next()) {
                         id = generatedKeys.getLong(1);
-                        utility.logWithDate("------ Generated " + type + " " + id + " --------------");
-                    }
-                } finally {
-                    try {
-                        generatedKeys.close();
-                    } catch (Exception e) {
-                        e.printStackTrace();
+                        log.trace("------ Generated {} {} --------------", type, id);
```

Comment:
Note: This log statement was the main cause of excessive logging, overwhelming the console.